github.com.conf.template

```diff
@@ -0,0 +1,31 @@
+# Proxy git's smart http protocol to only certain organisations on github.com.
+#
+# https://git-scm.com/docs/http-protocol
+#
+# The git http protocol design makes it non-trivial to distinguish between push
+# and pull - they are identical in terms of http verb and path.
+# 
+# We are keen to block push, but in practice we still need so support it for
+# now. We may be able to limit it by restricting the client_max_body_size to
+# very small size in future.
+server {
+
+    server_name github-proxy.opensafely.org;
+    root /var/www/html/;
+    listen ${PORT};
+
+    # The default nginx request body size is 1M, we make it explicit and bump
+    # to support current use cases
+    client_max_body_size 2M;
+
+    # only opensafely and opensafely-core orgs can be accessed
+    location ~ ^/(opensafely|opensafely-core)/.*/(info|git-upload-pack|git-receive-pack) {
```

Comment:
hmmm, is there scope to spoof this regex? we want to accept urls such as `github-proxy.opensafely.org/opensafely-core/job-runner/info/refs?service=git-upload-pack` but what about urls such as `github-proxy.opensafely.org/opensafely-core/job-runner/info/refs?service=whatever-you-want&haxor=git-upload-pack`?

Reply:
I have attempted to improve the situation